src/main/java/gov/cms/madie/qdm/humanreadable/model/HumanReadableCodeModel.java

```diff
@@ -0,0 +1,167 @@
+package gov.cms.madie.qdm.humanreadable.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Objects;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class HumanReadableCodeModel implements HumanReadableTerminologyModel {
+  private String name;
+  private String oid;
+  private String codesystemName;
+  private String codesystemVersion;
+  private boolean isCodesystemVersionIncluded;
+  private String datatype;
+  private String terminologyDisplay;
+  private String dataCriteriaDisplay;
+
+  public HumanReadableCodeModel(
+      String name,
+      String oid,
+      String codesystemName,
+      boolean isCodeSystemVersionIncluded,
+      String codesystemVersion,
+      String datatype) {
+    this.name = name;
+    this.oid = oid;
+    this.codesystemName = codesystemName;
+    this.codesystemVersion = codesystemVersion;
+    this.isCodesystemVersionIncluded = isCodeSystemVersionIncluded;
+    this.datatype = datatype;
+    createDataCriteriaDisplay();
+    createTerminologyDisplay();
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public String getOid() {
+    return oid;
+  }
+
+  public void setOid(String oid) {
+    this.oid = oid;
+  }
+
+  public String getCodesystemName() {
+    return codesystemName;
+  }
+
+  public void setCodesystemName(String taxonomy) {
+    this.codesystemName = taxonomy;
+  }
+
+  public boolean getIsCodesystemVersionIncluded() {
+    return isCodesystemVersionIncluded;
+  }
+
+  public void setIsCodesystemVersionIncluded(boolean isCodeSystemIncluded) {
+    this.isCodesystemVersionIncluded = isCodeSystemIncluded;
+  }
```

Comment:
Boilerplate setters can be removed since they'll be added by lombak's `@Data` annotation.

Reply:
Done